TS-4377: update documentation links in default configuration files

jsime opened the pull request:
- Updates existing links in default configuration files to point to current Administrator's Guide documentation URLs.
- Adds links to TS documentation for all default configuration files which a) did not already have them, and b) which have documentation.

maskit commented on proxy/config/records.config.default.in:
```diff
@@ -15,12 +15,11 @@ CONFIG proxy.config.ssl.number.threads INT -1
 ##############################################################################
 # Specify server addresses and ports to bind for HTTP and HTTPS. Docs:
-#    https://docs.trafficserver.apache.org/records.config#proxy-config-http-server-ports
+#  i  https://docs.trafficserver.apache.org/records.config#proxy.config.http.server_ports
```
There is a superfluous "i" at the beginning.

maskit commented:
Some URLs are shorten. They should be unified?

e.g.
https://docs.trafficserver.apache.org/records.config#thread-variables
https://docs.trafficserver.apache.org/en/latest/admin-guide/files/records.config.en.html

jsime commented:
Thanks for the catches. Fixed that 'i' typo and I switched the full URL in the comment at the top of records.config.default.in over to the shortened version. The links to other config files' documentation pages still reference the full URLs because we don't have redirects set up for those on the docs host right now. That could be a useful thing to have, though.

maskit commented:
Looks good to me.

jpeach commented:
@jsime Ready to merge this one?
